app/src/lib/dispatcher/app-store.ts

```diff
@@ -1875,13 +1891,40 @@ export class AppStore {
     return this.refreshGitHubRepositoryInfo(repository)
   }
 
+  private async getAccountForRemoteURL(
+    remote: string
+  ): Promise<IGitAccount | null> {
+    const gitHubRepository = matchGitHubRepository(this.accounts, remote)
+    if (gitHubRepository) {
+      const account = getAccountForEndpoint(
+        this.accounts,
+        gitHubRepository.endpoint
+      )
+      if (account) {
+        return account
+      }
+    }
+
+    const hostname = getGenericHostname(remote)
+    const username = await getGenericUsername(hostname)
```

Comment:
Fair point. _Ideally_ (and this is me introducing some scope creep because :trollface:) you'd be able to view and clean up these accounts from in the Preferences UI. And maybe we can do this down the track as part of an update to the app, with it migrate the existing storage of accounts to whatever data structure fits with what we need.

I just wonder if having something in the app's localStorage now will help us achieve that.